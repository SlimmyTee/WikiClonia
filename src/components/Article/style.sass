@import "./../../styles/_typography.css"
@import "./../../styles/_variables.sass"

.article
  display: grid
  width: 100%
  margin: 0
  padding: 50px 0

  background-color: var(--bg-color)

.hero
  margin: 0
  width: 100%
  display: grid
  grid-template-columns: 10% 15% 50% 15% 10%
  position: relative
  * + *
    margin-bottom: 0.7em
  a
    color: var(--primary-color)
  sup a
    border: none
  &__title
    grid-column: 3 / span 1
    font-family: "Heading", "Georgia", "Times New Roman"
    font-size: $size-title
    text-align: center


  &__credit
    grid-column: 3 / span 1
    margin-top: 0.5rem
    font-family: "Menu", -apple-system, BlinkMacSystemFont
    font-size: $size0
    text-align: center
    color: var(--support-color)
    font-weight: 600
    // display: none

  .wiki-content
    grid-column: 3 / span 1
    *
      grid-column: 3 / span 1

@media all and (max-width: $mobile-brp)
  .hero
    grid-template-columns: 1% 1% 96% 1% 1%

