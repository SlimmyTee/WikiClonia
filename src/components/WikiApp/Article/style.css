.article {
  padding: 50px 0;
}
