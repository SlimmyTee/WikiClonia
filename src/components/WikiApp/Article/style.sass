@import "./../_typography.css"
@import "./../_variables.sass"

.article
  display: grid
  // grid-template-columns: 9% 2% 56% 5% 2% 24% 2%

  width: 100%
  margin: 0
  padding: 50rem 0

  background-color: var(--bg-color)

.hero
  margin: 0 5%
  display: grid
  grid-template-columns: $margin-left $space-left $main-left $main-mid $main-right $space-right $margin-right
  padding-top: $space3
  position: relative
  @media only screen and (min-width: 2000px)
    margin: 0 20vw

  * + *
    margin-bottom: 0.7em
  a
    color: var(--primary-color)
  sup a
    border: none
  &__title
    grid-column: 3 / span 3
    font-family: "Title", "Georgia", "Times New Roman"
    font-size: $size-heading0
    text-align: center

  &__credit
    grid-column: 3 / span 3
    margin-top: 0.5rem
    font-family: "Credit", "Georgia", "Times New Roman"
    font-size: $size2
    text-align: center
    color: #AEB7B3
    font-weight: 500

  .wiki-content
    grid-column: 3 / span 3
    // display: grid
    // grid-template-columns: 10% 2% 25% 26% 25% 2% 10%
    *
      grid-column: 3 / span 3
