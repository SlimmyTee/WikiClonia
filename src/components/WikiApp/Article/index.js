import React, { Fragment } from "react";
import Content from "./../Content";
import Navigation from "./../Navigation";
import {
  usePageContent,
  useImages,
  useMetaData
} from "./../../../hooks/useWiki.js";
import { SkeletonTheme } from "react-loading-skeleton";
import { useParams } from "react-router-dom";
import "lazysizes";
import "./style.sass";

// import { buildURL, revisionParams } from "./../../../WikiWrapper";

export const ImagesContext = React.createContext(null);

const Article = ({ force_title }) => {
  let title = force_title ? force_title : useParams()?.title;
  const pageContent = usePageContent(title);
  const images = useImages(title);
  const metaData = useMetaData(title);

  console.log(pageContent);
  React.useEffect(() => {
    let toggle = document.getElementById("theme-switch");

    let switcher = function(e) {
      e.preventDefault();
      if (document.body.classList.contains("funky")) {
        toggle.innerText = "LIGHT MODE";
        document.body.classList.remove("funky");
      } else {
        toggle.innerText = "DARK MODE";
        document.body.classList.add("funky");
      }
    };
    toggle?.addEventListener("click", switcher);

    return () => toggle.removeEventListener("click", switcher);
  }, []);

  return (
    <ImagesContext.Provider value={{ images }}>
      <Fragment>
        <SkeletonTheme color="transparent">
          <div className="article">
            <div className="hero">
              <div className="hero__title">{title.replace(/_/g, " ")}</div>
              <div className="hero__credit">
                {/*From Wikipedia, the free encyclopedia*/}
                {`Created on ${metaData.date} by ${metaData.creator} | ${(
                  pageContent?.wordCount / 200
                )?.toFixed(0)} min read`.toUpperCase()}
              </div>
              <Content content={pageContent?.children} />
            </div>
          </div>
        </SkeletonTheme>
        <Navigation headings={pageContent?.headings} />
      </Fragment>
    </ImagesContext.Provider>
  );
};

export default Article;
