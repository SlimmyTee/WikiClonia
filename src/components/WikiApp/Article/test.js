import React from "react";
import Article from "./";
import renderer from "react-test-renderer";

const titles = [
  "Pet_door",
  "New_York_City",
  "The_Last_Supper_(Leonardo)",
  "Leonardo_da_Vinci",
  "Mona_Lisa",
  "Renaissance"
];

for (const title of titles) {
  it("renders correctly with " + title, () => {
    const tree = renderer.create(<Article title={title} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
}
