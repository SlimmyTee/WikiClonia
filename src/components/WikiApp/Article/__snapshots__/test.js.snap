// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with Leonardo_da_Vinci 1`] = `
Array [
  <div
    className="css-1lqxkvd-SkeletonTheme"
  >
    <div
      className="article"
    >
      <div
        className="hero"
      >
        <div
          className="hero__title"
        >
          Leonardo da Vinci
        </div>
        <div
          className="hero__credit"
        >
          CREATED ON NULL BY NULL | NAN MIN READ
        </div>
        <div
          className="wiki-content"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>,
  <div
    className="navigation-wrapper"
  >
    <div
      className="navigation"
    >
      <div
        className="navigation__content"
      >
        <button
          className="navigation__button navigation__left-button"
          onClick={[Function]}
        >
          <test-file-stub />
        </button>
        <div
          className="navigation__headings"
        />
        <button
          className="navigation__button navigation__right-button"
          onClick={[Function]}
        >
          <test-file-stub />
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`renders correctly with Mona_Lisa 1`] = `
Array [
  <div
    className="css-1lqxkvd-SkeletonTheme"
  >
    <div
      className="article"
    >
      <div
        className="hero"
      >
        <div
          className="hero__title"
        >
          Mona Lisa
        </div>
        <div
          className="hero__credit"
        >
          CREATED ON NULL BY NULL | NAN MIN READ
        </div>
        <div
          className="wiki-content"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>,
  <div
    className="navigation-wrapper"
  >
    <div
      className="navigation"
    >
      <div
        className="navigation__content"
      >
        <button
          className="navigation__button navigation__left-button"
          onClick={[Function]}
        >
          <test-file-stub />
        </button>
        <div
          className="navigation__headings"
        />
        <button
          className="navigation__button navigation__right-button"
          onClick={[Function]}
        >
          <test-file-stub />
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`renders correctly with New_York_City 1`] = `
Array [
  <div
    className="css-1lqxkvd-SkeletonTheme"
  >
    <div
      className="article"
    >
      <div
        className="hero"
      >
        <div
          className="hero__title"
        >
          New York City
        </div>
        <div
          className="hero__credit"
        >
          CREATED ON NULL BY NULL | NAN MIN READ
        </div>
        <div
          className="wiki-content"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>,
  <div
    className="navigation-wrapper"
  >
    <div
      className="navigation"
    >
      <div
        className="navigation__content"
      >
        <button
          className="navigation__button navigation__left-button"
          onClick={[Function]}
        >
          <test-file-stub />
        </button>
        <div
          className="navigation__headings"
        />
        <button
          className="navigation__button navigation__right-button"
          onClick={[Function]}
        >
          <test-file-stub />
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`renders correctly with Pet_door 1`] = `
Array [
  <div
    className="css-1lqxkvd-SkeletonTheme"
  >
    <div
      className="article"
    >
      <div
        className="hero"
      >
        <div
          className="hero__title"
        >
          Pet door
        </div>
        <div
          className="hero__credit"
        >
          CREATED ON NULL BY NULL | NAN MIN READ
        </div>
        <div
          className="wiki-content"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>,
  <div
    className="navigation-wrapper"
  >
    <div
      className="navigation"
    >
      <div
        className="navigation__content"
      >
        <button
          className="navigation__button navigation__left-button"
          onClick={[Function]}
        >
          <test-file-stub />
        </button>
        <div
          className="navigation__headings"
        />
        <button
          className="navigation__button navigation__right-button"
          onClick={[Function]}
        >
          <test-file-stub />
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`renders correctly with Renaissance 1`] = `
Array [
  <div
    className="css-1lqxkvd-SkeletonTheme"
  >
    <div
      className="article"
    >
      <div
        className="hero"
      >
        <div
          className="hero__title"
        >
          Renaissance
        </div>
        <div
          className="hero__credit"
        >
          CREATED ON NULL BY NULL | NAN MIN READ
        </div>
        <div
          className="wiki-content"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>,
  <div
    className="navigation-wrapper"
  >
    <div
      className="navigation"
    >
      <div
        className="navigation__content"
      >
        <button
          className="navigation__button navigation__left-button"
          onClick={[Function]}
        >
          <test-file-stub />
        </button>
        <div
          className="navigation__headings"
        />
        <button
          className="navigation__button navigation__right-button"
          onClick={[Function]}
        >
          <test-file-stub />
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`renders correctly with The_Last_Supper_(Leonardo) 1`] = `
Array [
  <div
    className="css-1lqxkvd-SkeletonTheme"
  >
    <div
      className="article"
    >
      <div
        className="hero"
      >
        <div
          className="hero__title"
        >
          The Last Supper (Leonardo)
        </div>
        <div
          className="hero__credit"
        >
          CREATED ON NULL BY NULL | NAN MIN READ
        </div>
        <div
          className="wiki-content"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>,
  <div
    className="navigation-wrapper"
  >
    <div
      className="navigation"
    >
      <div
        className="navigation__content"
      >
        <button
          className="navigation__button navigation__left-button"
          onClick={[Function]}
        >
          <test-file-stub />
        </button>
        <div
          className="navigation__headings"
        />
        <button
          className="navigation__button navigation__right-button"
          onClick={[Function]}
        >
          <test-file-stub />
        </button>
      </div>
    </div>
  </div>,
]
`;
